workflow/metrics/labels.go

```diff
@@ -1,6 +1,15 @@
 package metrics
 
 const (
+	labelBuildVersion      string = `version`
+	labelBuildPlatform     string = `platform`
+	labelBuildGoVersion    string = `goversion`
```

Comment:
```suggestion
	labelBuildGoVersion    string = `go_version`
```

Split words with underscores, per OTel convention per https://github.com/argoproj/argo-workflows/pull/13269#discussion_r1661378969 and https://github.com/argoproj/argo-workflows/pull/13269#discussion_r1661397992